detectors/gcp/cloud-function.go

```diff
@@ -0,0 +1,70 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package gcp
+
+import (
+	"context"
+	"os"
+
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/sdk/resource"
+	semconv "go.opentelemetry.io/otel/semconv/v1.4.0"
+)
+
+const (
+	gcpFunctionNameKey = "K_SERVICE"
+)
+
+//NewCloudFunction will return an implementation for gcp cloud function resource detector
```

Comment:
suggetion includes:

- Prepend space
- Use capitalized form of proper noun
- End complete sentence with punctuation

```suggestion
// NewCloudFunction will return a GCP Cloud Function resource detector.
```

Reply:
Done